### PR TITLE
python3Packages.fiblary3: init at 0.1.12

### PR DESCRIPTION
**Pull request by graham33**
###### Motivation for this change
Library for Fibaro devices (used in Home Assistant).

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/fiblary3/default.nix
Comment (AndersonTorres) [inline pkgs/development/python-modules/fiblary3/default.nix:41]: I would merely suggest to swap license and description here.

It keeps the attributes in alphabetical order. 
The exception is homepage, because it is the most important attribute!

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 114049` at df225f86 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/86f5e9a1c3077b54c5afb42264aaf249">python37Packages.fiblary3</a></li></li>
    <li><a href="https://gist.github.com/8cd85b03a0a7f4aaf70180bab079c4cb">python38Packages.fiblary3</a></li></li>
    <li><a href="https://gist.github.com/1200463237916b5e11458483752af42d">python39Packages.fiblary3</a></li></li>
  </ul>
</details>

File: pkgs/servers/home-assistant/component-packages.nix
Comment (AndersonTorres) [inline pkgs/servers/home-assistant/component-packages.nix:251]: Put this in a different commit, because they are about different things even if related.

(It is necessary because of future cherry-pickings and rebases.)

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/fiblary3/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/fiblary3/default.nix:23]:
```suggestion
```

File: pkgs/development/python-modules/fiblary3/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/fiblary3/default.nix:16]:
```suggestion
  version = "0.1.8";
  disabled = !isPy3k;
```

File: pkgs/development/python-modules/fiblary3/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/fiblary3/default.nix:41]:
```suggestion
    description = "Fibaro Home Center API Python Library";
    homepage = "https://github.com/pbalogh77/fiblary";
    license = licenses.asl20;
```

this is a more standard ordering used by many packages but tbh it does not matter.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/fiblary3/default.nix
Comment (AndersonTorres) [reply pkgs/development/python-modules/fiblary3/default.nix:41]: Homepage is better as first argument, followed by description + longDescription, and the others being alphabetically sorted. This is the most natural reading.

Comment (SuperSandro2000) [reply pkgs/development/python-modules/fiblary3/default.nix:41]: Doesn't matter to be honest. I'm fine with both.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 114049` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>python37Packages.fiblary3</li>
<li>python38Packages.fiblary3</li>
<li>python39Packages.fiblary3</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
python37Packages.fiblary3:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.


Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
python38Packages.fiblary3:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.


Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
python39Packages.fiblary3:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.


Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 114049` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>python37Packages.fiblary3</li>
<li>python38Packages.fiblary3</li>
<li>python39Packages.fiblary3</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
python37Packages.fiblary3:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.


Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
python38Packages.fiblary3:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.


Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
python39Packages.fiblary3:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.


Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
</details>

**Comment by graham33** [comment]
Thanks, I've made the suggested changes.

File: pkgs/development/python-modules/fiblary3/default.nix
Comment (fabaff) [inline pkgs/development/python-modules/fiblary3/default.nix:25]: `sphinx` is used to build the documentation. Is it really a build requirement?

**Review by fabaff — COMMENTED** [review]
The upstream repo contains [tests](https://github.com/pbalogh77/fiblary/tree/master/fiblary3/tests). If possible should those tests run during the build. 

If not, please add at least `pythonImportsCheck`. Thanks.

**Comment by graham33** [comment]
> The upstream repo contains [tests](https://github.com/pbalogh77/fiblary/tree/master/fiblary3/tests). If possible should those tests run during the build.

I did try this before, and it looks like the tests are excluded from the PyPI distribution (I'm guessing because the test package is excluded from setup.py).  Is this a common issue?  I could build from GitHub instead, but the repo doesn't have tags for any recent releases.

> If not, please add at least `pythonImportsCheck`. Thanks.

Done, thanks.

File: pkgs/development/python-modules/fiblary3/default.nix
Comment (graham33) [reply pkgs/development/python-modules/fiblary3/default.nix:25]: It's mentioned in [requirements.txt](https://github.com/pbalogh77/fiblary/blob/master/requirements.txt) and the build fails without it.  I don't think it's actually used unfortunately.

**Comment by SuperSandro2000** [comment]
> Is this a common issue?

Yes.

> but the repo doesn't have tags for any recent releases.

Could you request tags upstream?

**Comment by graham33** [comment]
> Could you request tags upstream?

The upstream repo doesn't have Issues enabled :)  It looks like the tests have problems, I will see if I can fix any of them and raise a PR, and ask about tags in that PR too.

**Comment by graham33** [comment]
> > Could you request tags upstream?
> 
> The upstream repo doesn't have Issues enabled :) It looks like the tests have problems, I will see if I can fix any of them and raise a PR, and ask about tags in that PR too.

It seemed worth putting some effort in since I have a sizeable dependency on this library, so I fixed the tests: https://github.com/pbalogh77/fiblary/pull/2.  They all now pass from the nix package too if I refer to my branch, so hopefully we can get this merged upstream.  Unfortunately it doesn't look like upstream has any CI so the tests had rotted quite badly.  I also asked about tags in the same PR.

File: pkgs/development/python-modules/fiblary3/default.nix
Comment (SuperSandro2000) [reply pkgs/development/python-modules/fiblary3/default.nix:25]: If it is unused we can patch it out.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 114049` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>python37Packages.fiblary3</li>
<li>python38Packages.fiblary3</li>
<li>python39Packages.fiblary3</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
python37Packages.fiblary3:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python38Packages.fiblary3:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python39Packages.fiblary3:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 114049` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>python37Packages.fiblary3</li>
<li>python38Packages.fiblary3</li>
<li>python39Packages.fiblary3</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
python37Packages.fiblary3:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python38Packages.fiblary3:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python39Packages.fiblary3:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>
</details>

**Comment by graham33** [comment]
I think until the above PR is merged upstream and or there is some sign of activity from the maintainer, it's likely best to hold off on merging this into nixpkgs.  Let me know if it's best to withdraw the PR or somehow mark it as a work in progress.

**Comment by graham33** [comment]
> > @graham33 I think you should fork fiblary and make upstream Home Assistant use your fork (in which the tests would be fixed).
> 
> It has been almost a year since I raised that PR, and no sign of activity on pbalogh77/fiblary since 2020. I can give it a try!

I've created a fork at https://github.com/graham33/fiblary and https://pypi.org/project/fiblary3-fork/ and updated this PR to use it.  Will create an issue in home-assistant upstream.

File: pkgs/development/python-modules/fiblary3/default.nix
Comment (graham33) [reply pkgs/development/python-modules/fiblary3/default.nix]: Fixed in my fork.

File: pkgs/top-level/python-packages.nix
Comment (dotlambda) [inline pkgs/top-level/python-packages.nix:2820]:
```suggestion
  fiblary3-fork = callPackage ../development/python-modules/fiblary3-fork { };
```
Let's add `fiblary3` as an alias.

File: pkgs/development/python-modules/fiblary3/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/fiblary3/default.nix:19]:
```suggestion
  pname = "fiblary3-fork";
```

File: pkgs/development/python-modules/fiblary3/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/fiblary3/default.nix:25]:
```suggestion
    inherit pname version;
```

File: pkgs/development/python-modules/fiblary3/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/fiblary3/default.nix:21]:
```suggestion
```

or use pythonOlder because python2 moved out recently.

File: pkgs/top-level/python-packages.nix
Comment (dotlambda) [inline pkgs/top-level/python-packages.nix:2836]: This should be in `python-aliases.nix`.

**Review by dotlambda — CHANGES_REQUESTED** [review]
You have to use `parse-requirements.py`.
For that to work you need to add an entry to `PKG_PREFERENCES`.

File: pkgs/top-level/python-packages.nix
Comment (dotlambda) [reply pkgs/top-level/python-packages.nix]: Actually I just removed it, there was no need for it.

**Review by dotlambda — APPROVED** [review]

**Comment by dotlambda** [comment]
Result of `nixpkgs-review pr 114049` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.fiblary3-fork</li>
    <li>python39Packages.fiblary3-fork</li>
  </ul>
</details>

**Comment by dotlambda** [comment]
@mweinelt Please take a look at the changes to `parse-requirements.py`.

**Comment by mweinelt** [comment]
Refactor of `parse-requirements.py` looks fine.

**Pull request merged**
